File: lib/wedjets/sentir_mod_wedjet.dart
import 'package:flutter/material.dart';
import '../yutelez/heksagon_djeyometre.dart';
import 'heksagon_wedjet.dart';

class SentirModWedjet extends StatelessWidget {
  final HeksagonDjeyometre geometry;
  final ValueChanged<bool>? onPressedChanged;
  final VoidCallback? onTap;
  final VoidCallback? onLongPress;
  final Color backgroundColor;
  final Color textColor;
  final String label;
  final Function(bool)? onHover;
  final Widget? child; // Add this line

  const SentirModWedjet({
    super.key,
    required this.geometry,
    this.onPressedChanged,
    this.onTap,
    this.onLongPress,
    required this.backgroundColor,
    required this.textColor,
    this.label = '',
    this.onHover,
    this.child, // Add this line
  });

  @override
  Widget build(BuildContext context) {
    return HeksagonWedjet(
      label: label,
      backgroundColor: backgroundColor,
      textColor: textColor,
      size: geometry.heksWidlx,
      rotationAngle: geometry.roteyconAngol,
      onTap: onTap,
      onLongPress: onLongPress,
      onPressedChanged: onPressedChanged,
      onHover: onHover,
      child: child, // Pass the child here
    );
  }
}

File: android/ime/src/main/kotlin/wedjets/SentirModWedjet.kt
package wedjets

import androidx.compose.runtime.Composable
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.platform.LocalDensity
import yuteledez.HeksagonDjeyometre

/**
 * A wrapper composable for creating the central hexagon widget.
 *
 * It simplifies the creation of the main hexagon by pre-configuring it
 * with properties derived from the HeksagonDjeyometre.
 */
@Composable
fun SentirModWedjet(
    geometry: HeksagonDjeyometre,
    onPressedChanged: ((Boolean) -> Unit)? = null,
    onTap: (() -> Unit)? = null,
    onLongPress: (() -> Unit)? = null,
    backgroundColor: Color,
    textColor: Color,
    label: String = "",
    onHover: ((Boolean) -> Unit)? = null,
    child: @Composable (() -> Unit)? = null
) {
    val density = LocalDensity.current
    val sizeInDp = with(density) { geometry.heksWidlx.toFloat().toDp() }

    HeksagonWedjet(
        label = label,
        backgroundColor = backgroundColor,
        textColor = textColor,
        size = sizeInDp,
        rotationAngle = geometry.roteyconAngol.toFloat(),
        onTap = onTap,
        onLongPress = onLongPress,
        onPressedChanged = onPressedChanged,
        onHover = onHover,
        child = child
    )
}
